sql/core/src/main/scala/org/apache/spark/sql/catalog/Catalog.scala

```diff
@@ -226,4 +226,11 @@ abstract class Catalog {
    */
   def refreshTable(tableName: String): Unit
 
+  /**
+   * Invalidate and refresh all the cached data (and the associated metadata) for any dataframe that
+   * contains the given data source path.
+   *
+   * @since 2.0.0
+   */
+  def refreshResource(path: String): Unit
```

Comment:
I like `invalidateCache()` but the reason for choosing `refreshResource()` was to make it sound similar to `refreshTable ()` above. Let me know if you prefer one over the other.